team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
> I pushed a fix: 0760697

Did you not also mean to add `if (!fromSelf && dexEntry.prevo === templateid) continue;` ? Because otherwise it would still not work properly for some cases, like the one with Scyther.

I've added that line to my latest commit and things seem to be working as expected.